tools/dcm-getframe.c

```diff
@@ -0,0 +1,152 @@
+#define _CRT_SECURE_NO_WARNINGS
+#define _CRT_NONSTDC_NO_DEPRECATE
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <dicom/dicom.h>
+
+
+static const char usage[] = "usage: dcm-getframe [-v] [-V] [-h] [-o OUTPUT-FILE] FILE_PATH FRAME_NUMBER\n";
+
+
+int main(int argc, char *argv[]) 
+{
+    DcmError *error = NULL;
+    char *output_file = NULL;
+    int i;
+
+    dcm_log_level = DCM_LOG_ERROR;
+
+    for (i = 1; i < argc && argv[i][0] == '-'; i++) {
+        switch (argv[i][1]) {
+            case 'h':
+                printf("%s\n", usage);
+                return EXIT_SUCCESS;
+            case 'V':
+                printf("%s\n", dcm_get_version());
+                return EXIT_SUCCESS;
+            case 'v':
+                dcm_log_level = DCM_LOG_INFO;
+                break;
+            case 'o':
+                output_file = argv[i + 1];
+                i += 1;
+                break;
+            default:
+                fprintf(stderr, "%s\n", usage);
+                return EXIT_FAILURE;
+        }
+    }
+
+    if ((i + 2) != argc) {
+        fprintf(stderr, "%s\n", usage);
+        return EXIT_FAILURE;
+    }
+    const char *file_path = argv[i];
+
+    uint32_t frame_number = atoi(argv[i + 1]);
+
+    dcm_log_info("Read file '%s'", file_path);
+    DcmFile *file = dcm_file_open(&error, file_path);
+    if (file == NULL) {
+        dcm_error_log(error);
+        dcm_error_clear(&error);
+        return EXIT_FAILURE;
+    }
+
+    dcm_log_info("Read metadata");
+    DcmDataSet *metadata = dcm_file_read_metadata(&error, file);
+    if (metadata == NULL) {
+        dcm_error_log(error);
+        dcm_error_clear(&error);
+        dcm_file_destroy(file);
+        return EXIT_FAILURE;
+    }
+
+    dcm_log_info("Read BOT");
+    DcmBOT *bot = dcm_file_read_bot(&error, file, metadata);
+    if (bot == NULL) {
+        dcm_error_log(error);
+        dcm_error_clear(&error);
+        dcm_dataset_destroy(metadata);
+        dcm_file_destroy(file);
+        return EXIT_FAILURE;
+    }
```

Comment:
If no BOT is found, the program should try to build one using `dcm_file_build_bot` rather than fail.

Reply:
ok, done!